Error when pip3 install -r requirements.txt on macos 11.1 with M1 chip

### What is the question?

Something wrong happened when executing `pip3 install -r requirements.txt` on macos 11.1 with M1 chip.

### Any additional information?

```
  ERROR: Command errored out with exit status 1:
   command: /Library/Developer/CommandLineTools/usr/bin/python3 /Users/zhangzheng/Library/Python/3.8/lib/python/site-packages/pip/_vendor/pep517/_in_process.py build_wheel /var/folders/lv/vdwc066d2531zy7px8mrh2jm0000gn/T/tmpwvyg6shg
       cwd: /private/var/folders/lv/vdwc066d2531zy7px8mrh2jm0000gn/T/pip-install-jo9vwzeb/cryptography_0956c6893d8b47a5abeddfc5fc1b37d5
  Complete output (147 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.macosx-10.14.6-arm64-3.8
  creating build/lib.macosx-10.14.6-arm64-3.8/cryptography
  copying src/cryptography/__init__.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography
  copying src/cryptography/utils.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography
  copying src/cryptography/__about__.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography
  copying src/cryptography/exceptions.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography
  copying src/cryptography/fernet.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography
  creating build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat
  copying src/cryptography/hazmat/_der.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat
  copying src/cryptography/hazmat/__init__.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat
  copying src/cryptography/hazmat/_oid.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat
  creating build/lib.macosx-10.14.6-arm64-3.8/cryptography/x509
  copying src/cryptography/x509/oid.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/x509
  copying src/cryptography/x509/ocsp.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/x509
  copying src/cryptography/x509/general_name.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/x509
  copying src/cryptography/x509/__init__.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/x509
  copying src/cryptography/x509/extensions.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/x509
  copying src/cryptography/x509/name.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/x509
  copying src/cryptography/x509/base.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/x509
  copying src/cryptography/x509/certificate_transparency.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/x509
  creating build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/backends
  copying src/cryptography/hazmat/backends/interfaces.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/backends
  copying src/cryptography/hazmat/backends/__init__.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/backends
  creating build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives
  copying src/cryptography/hazmat/primitives/cmac.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives
  copying src/cryptography/hazmat/primitives/__init__.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives
  copying src/cryptography/hazmat/primitives/poly1305.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives
  copying src/cryptography/hazmat/primitives/constant_time.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives
  copying src/cryptography/hazmat/primitives/keywrap.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives
  copying src/cryptography/hazmat/primitives/hmac.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives
  copying src/cryptography/hazmat/primitives/hashes.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives
  copying src/cryptography/hazmat/primitives/padding.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives
  creating build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/bindings
  copying src/cryptography/hazmat/bindings/__init__.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/bindings
  creating build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/x448.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/backend.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/ec.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/ciphers.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/ocsp.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/x509.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/aead.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/encode_asn1.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/rsa.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/dh.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/cmac.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/__init__.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/utils.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/poly1305.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/ed25519.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/dsa.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/decode_asn1.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/hmac.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/ed448.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/x25519.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/backends/openssl
  copying src/cryptography/hazmat/backends/openssl/hashes.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/backends/openssl
  creating build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives/kdf
  copying src/cryptography/hazmat/primitives/kdf/__init__.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives/kdf
  copying src/cryptography/hazmat/primitives/kdf/scrypt.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives/kdf
  copying src/cryptography/hazmat/primitives/kdf/pbkdf2.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives/kdf
  copying src/cryptography/hazmat/primitives/kdf/hkdf.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives/kdf
  copying src/cryptography/hazmat/primitives/kdf/x963kdf.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives/kdf
  copying src/cryptography/hazmat/primitives/kdf/kbkdf.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives/kdf
  copying src/cryptography/hazmat/primitives/kdf/concatkdf.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives/kdf
  creating build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives/twofactor
  copying src/cryptography/hazmat/primitives/twofactor/totp.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives/twofactor
  copying src/cryptography/hazmat/primitives/twofactor/__init__.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives/twofactor
  copying src/cryptography/hazmat/primitives/twofactor/hotp.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives/twofactor
  copying src/cryptography/hazmat/primitives/twofactor/utils.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives/twofactor
  creating build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives/serialization
  copying src/cryptography/hazmat/primitives/serialization/__init__.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives/serialization
  copying src/cryptography/hazmat/primitives/serialization/pkcs12.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives/serialization
  copying src/cryptography/hazmat/primitives/serialization/pkcs7.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives/serialization
  copying src/cryptography/hazmat/primitives/serialization/ssh.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives/serialization
  copying src/cryptography/hazmat/primitives/serialization/base.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives/serialization
  creating build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives/ciphers
  copying src/cryptography/hazmat/primitives/ciphers/algorithms.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives/ciphers
  copying src/cryptography/hazmat/primitives/ciphers/aead.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives/ciphers
  copying src/cryptography/hazmat/primitives/ciphers/__init__.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives/ciphers
  copying src/cryptography/hazmat/primitives/ciphers/modes.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives/ciphers
  copying src/cryptography/hazmat/primitives/ciphers/base.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives/ciphers
  creating build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives/asymmetric
  copying src/cryptography/hazmat/primitives/asymmetric/x448.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives/asymmetric
  copying src/cryptography/hazmat/primitives/asymmetric/ec.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives/asymmetric
  copying src/cryptography/hazmat/primitives/asymmetric/rsa.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives/asymmetric
  copying src/cryptography/hazmat/primitives/asymmetric/dh.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives/asymmetric
  copying src/cryptography/hazmat/primitives/asymmetric/__init__.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives/asymmetric
  copying src/cryptography/hazmat/primitives/asymmetric/utils.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives/asymmetric
  copying src/cryptography/hazmat/primitives/asymmetric/ed25519.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives/asymmetric
  copying src/cryptography/hazmat/primitives/asymmetric/dsa.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives/asymmetric
  copying src/cryptography/hazmat/primitives/asymmetric/ed448.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives/asymmetric
  copying src/cryptography/hazmat/primitives/asymmetric/x25519.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives/asymmetric
  copying src/cryptography/hazmat/primitives/asymmetric/padding.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/primitives/asymmetric
  creating build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/bindings/openssl
  copying src/cryptography/hazmat/bindings/openssl/__init__.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/bindings/openssl
  copying src/cryptography/hazmat/bindings/openssl/_conditional.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/bindings/openssl
  copying src/cryptography/hazmat/bindings/openssl/binding.py -> build/lib.macosx-10.14.6-arm64-3.8/cryptography/hazmat/bindings/openssl
  running egg_info
  writing src/cryptography.egg-info/PKG-INFO
  writing dependency_links to src/cryptography.egg-info/dependency_links.txt
  writing requirements to src/cryptography.egg-info/requires.txt
  writing top-level names to src/cryptography.egg-info/top_level.txt
  reading manifest file 'src/cryptography.egg-info/SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  no previously-included directories found matching 'docs/_build'
  warning: no previously-included files found matching 'vectors'
  warning: no previously-included files matching '*' found under directory 'vectors'
  warning: no previously-included files matching '*' found under directory '.github'
  warning: no previously-included files found matching 'release.py'
  warning: no previously-included files found matching '.coveragerc'
  warning: no previously-included files found matching 'codecov.yml'
  warning: no previously-included files found matching '.readthedocs.yml'
  warning: no previously-included files found matching 'dev-requirements.txt'
  warning: no previously-included files found matching 'rtd-requirements.txt'
  warning: no previously-included files found matching 'tox.ini'
  warning: no previously-included files matching '*' found under directory '.zuul.d'
  warning: no previously-included files matching '*' found under directory '.zuul.playbooks'
  writing manifest file 'src/cryptography.egg-info/SOURCES.txt'
  running build_ext
  generating cffi module 'build/temp.macosx-10.14.6-arm64-3.8/_padding.c'
  creating build/temp.macosx-10.14.6-arm64-3.8
  generating cffi module 'build/temp.macosx-10.14.6-arm64-3.8/_openssl.c'
  building '_openssl' extension
  creating build/temp.macosx-10.14.6-arm64-3.8/build
  creating build/temp.macosx-10.14.6-arm64-3.8/build/temp.macosx-10.14.6-arm64-3.8
  clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -iwithsysroot/System/Library/Frameworks/System.framework/PrivateHeaders -iwithsysroot/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.8/Headers -arch arm64 -arch x86_64 -I/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.8/include/python3.8 -c build/temp.macosx-10.14.6-arm64-3.8/_openssl.c -o build/temp.macosx-10.14.6-arm64-3.8/build/temp.macosx-10.14.6-arm64-3.8/_openssl.o -Wconversion -Wno-error=sign-conversion
  build/temp.macosx-10.14.6-arm64-3.8/_openssl.c:575:10: fatal error: 'openssl/opensslv.h' file not found
  #include <openssl/opensslv.h>
           ^~~~~~~~~~~~~~~~~~~~
  1 error generated.
  
      =============================DEBUG ASSISTANCE=============================
      If you are seeing a compilation error please try the following steps to
      successfully install cryptography:
      1) Upgrade to the latest pip and try again. This will fix errors for most
         users. See: https://pip.pypa.io/en/stable/installing/#upgrading-pip
      2) Read https://cryptography.io/en/latest/installation.html for specific
         instructions for your platform.
      3) Check our frequently asked questions for more information:
         https://cryptography.io/en/latest/faq.html
      =============================DEBUG ASSISTANCE=============================
  
  error: command 'clang' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for cryptography
ERROR: Could not build wheels for cryptography which use PEP 517 and cannot be installed directly
```

#### Checker:
 - [x] I read the documentation (README)

--- a/default.conf
+++ b/default.conf
@@ -6,7 +6,7 @@ force-extensions = False
 [general]
 threads = 30
 recursive = False
-recursive-level-max = 0
+recursion-depth = 0
 exclude-subdirs = %%ff/
 random-user-agents = False
 save-logs-home = False
@@ -17,8 +17,8 @@ color = True
 # exclude-status = 500,502
 # exclude-sizes = 0b,123gb
 # exclude-texts = "Not found"
-# exclude-regexps = Forbidde[a-z]{1}
-# calibration-path = InvalidPath123123
+# exclude-regexps = 403 [a-z]{1,25}
+# calibration-path = InvalidPath123
 
 [reports]
 autosave-report = True

--- a/lib/connection/requester.py
+++ b/lib/connection/requester.py
@@ -124,7 +124,7 @@ class Requester(object):
         )
 
     def setHeader(self, key, value):
-        self.headers[key.strip()] = value.strip()
+        self.headers[key.strip()] = value.strip() if value else value
 
     def setRandomAgents(self, agents):
         self.randomAgents = list(agents)

--- a/lib/controller/controller.py
+++ b/lib/controller/controller.py
@@ -63,22 +63,7 @@ class Controller(object):
         self.savePath = self.script_path
         self.doneDirs = []
 
-        if arguments.urlList:
-            default_headers = {
-                "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36",
-                "Accept-Language": "*",
-                "Accept-Encoding": "*",
-                "Keep-Alive": "300",
-                "Cache-Control": "max-age=0",
-            }
-
-            self.urlList = list(filter(None, dict.fromkeys(arguments.urlList)))
-            self.httpmethod = arguments.httpmethod.lower()
-            self.data = arguments.data
-            self.headers = {**default_headers, **arguments.headers}
-            self.cookie = arguments.cookie
-            self.useragent = arguments.useragent
-        else:
+        if arguments.raw_file:
             # Overwrite python-requests default headers
             default_headers = {
                 "User-Agent": None,
@@ -93,6 +78,21 @@ class Controller(object):
             self.headers = {**default_headers, **_raw.headers()}
             self.cookie = _raw.cookie()
             self.useragent = _raw.user_agent()
+        else:
+            default_headers = {
+                "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36",
+                "Accept-Language": "*",
+                "Accept-Encoding": "*",
+                "Keep-Alive": "300",
+                "Cache-Control": "max-age=0",
+            }
+
+            self.urlList = list(filter(None, dict.fromkeys(arguments.urlList)))
+            self.httpmethod = arguments.httpmethod.lower()
+            self.data = arguments.data
+            self.headers = {**default_headers, **arguments.headers}
+            self.cookie = arguments.cookie
+            self.useragent = arguments.useragent
 
         self.recursion_depth = arguments.recursion_depth
 
@@ -268,7 +268,6 @@ class Controller(object):
         self.output.warning("\nTask Completed")
 
     def printConfig(self):
-
         self.output.config(
             ', '.join(self.arguments.extensions),
             ', '.join(self.arguments.prefixes),

--- a/lib/controller/controller.py
+++ b/lib/controller/controller.py
@@ -94,7 +94,7 @@ class Controller(object):
             self.cookie = _raw.cookie()
             self.useragent = _raw.user_agent()
 
-        self.recursive_level_max = arguments.recursive_level_max
+        self.recursion_depth = arguments.recursion_depth
 
         if arguments.saveHome:
             savePath = self.getSavePath()
@@ -573,8 +573,8 @@ class Controller(object):
                 path.path, path.response, self.arguments.full_url, addedToQueue
             )
 
-            if self.arguments.matches_proxy:
-                self.requester.request(path.path, proxy=self.arguments.matches_proxy)
+            if self.arguments.replay_proxy:
+                self.requester.request(path.path, proxy=self.arguments.replay_proxy)
 
             newPath = self.currentDirectory + path.path
 
@@ -707,7 +707,7 @@ class Controller(object):
                 return False
             elif dir in self.doneDirs:
                 return False
-            elif self.recursive_level_max and dir.count("/") > self.recursive_level_max:
+            elif self.recursive_depth and dir.count("/") > self.recursion_depth:
                 return False
 
             self.directories.put(dir)
@@ -742,7 +742,7 @@ class Controller(object):
                 return False
             elif dir in self.doneDirs:
                 return False
-            elif self.recursive_level_max and dir.count("/") > self.recursive_level_max:
+            elif self.recursion_depth and dir.count("/") > self.recursion_depth:
                 return False
 
             self.directories.put(dir)

--- a/lib/controller/controller.py
+++ b/lib/controller/controller.py
@@ -65,7 +65,7 @@ class Controller(object):
 
         if arguments.urlList:
             default_headers = {
-                "User-agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36",
+                "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36",
                 "Accept-Language": "*",
                 "Accept-Encoding": "*",
                 "Keep-Alive": "300",
@@ -81,7 +81,7 @@ class Controller(object):
         else:
             # Overwrite python-requests default headers
             default_headers = {
-                "User-agent": None,
+                "User-Agent": None,
                 "Accept-Encoding": None,
                 "Accept": None,
             }

--- a/lib/core/argument_parser.py
+++ b/lib/core/argument_parser.py
@@ -384,10 +384,6 @@ class ArgumentParser(object):
             print("Invalid URI scheme: {0}".format(self.scheme))
             exit(1)
 
-        if self.scheme not in ["http", "https"]:
-            print("Invalid URI scheme: {0}".format(self.scheme))
-            exit(1)
-
     def parseConfig(self):
         config = DefaultConfigParser()
         configPath = FileUtils.build_path(self.script_path, "default.conf")

--- a/lib/core/argument_parser.py
+++ b/lib/core/argument_parser.py
@@ -136,12 +136,12 @@ class ArgumentParser(object):
         else:
             self.proxy = None
 
-        if options.matches_proxy:
-            self.matches_proxy = options.matches_proxy
+        if options.replay_proxy:
+            self.replay_proxy = options.replay_proxy
             options.requestByHostname = True
 
         else:
-            self.matches_proxy = None
+            self.replay_proxy = None
 
         if options.headers:
             try:
@@ -378,7 +378,11 @@ class ArgumentParser(object):
         self.exit_on_error = options.exit_on_error
         self.debug = options.debug
 
-        self.recursive_level_max = options.recursive_level_max
+        self.recursion_depth = options.recursion_depth
+
+        if self.scheme not in ["http", "https"]:
+            print("Invalid URI scheme: {0}".format(self.scheme))
+            exit(1)
 
         if self.scheme not in ["http", "https"]:
             print("Invalid URI scheme: {0}".format(self.scheme))
@@ -405,7 +409,7 @@ class ArgumentParser(object):
         self.excludeRegexps = config.safe_get("general", "exclude-regexps", None)
         self.excludeRedirects = config.safe_get("general", "exclude-redirects", None)
         self.recursive = config.safe_getboolean("general", "recursive", False)
-        self.recursive_level_max = config.safe_getint("general", "recursive-level-max", 0)
+        self.recursion_depth = config.safe_getint("general", "recursion-depth", 0)
         self.testFailPath = config.safe_get("general", "calibration-path", "").strip()
         self.saveHome = config.safe_getboolean("general", "save-logs-home", False)
         self.excludeSubdirs = config.safe_get("general", "exclude-subdirs", None)
@@ -450,7 +454,7 @@ class ArgumentParser(object):
         self.proxy = config.safe_get("connection", "proxy", None)
         self.proxylist = config.safe_get("connection", "proxy-list", None)
         self.scheme = config.safe_get("connection", "scheme", "http", ["http", "https"])
-        self.matches_proxy = config.safe_get("connection", "replay-proxy", None)
+        self.replay_proxy = config.safe_get("connection", "replay-proxy", None)
         self.requestByHostname = config.safe_getboolean(
             "connection", "request-by-hostname", False
         )
@@ -506,7 +510,7 @@ information at https://github.com/maurosoria/dirsearch.""")
         general.add_option("-r", "--recursive", help="Bruteforce recursively", action="store_true", dest="recursive",
                            default=self.recursive)
         general.add_option("-R", "--recursion-depth", help="Maximum recursion depth", action="store",
-                           type="int", dest="recursive_level_max", default=self.recursive_level_max, metavar="DEPTH")
+                           type="int", dest="recursion_depth", default=self.recursion_depth, metavar="DEPTH")
         general.add_option("-t", "--threads", help="Number of threads", action="store", type="int", dest="threadsCount",
                            default=self.threadsCount, metavar="THREADS")
         general.add_option("--subdirs", help="Scan sub-directories of the given URL[s] (separated by commas)", action="store",
@@ -568,7 +572,7 @@ information at https://github.com/maurosoria/dirsearch.""")
                               help="Proxy URL, support HTTP and SOCKS proxies (Example: localhost:8080, socks5://localhost:8088)", metavar="PROXY")
         connection.add_option("--proxy-list", action="store", dest="proxyList", type="string",
                               default=self.proxylist, help="File contains proxy servers", metavar="FILE")
-        connection.add_option("--replay-proxy", action="store", dest="matches_proxy", type="string", default=self.matches_proxy,
+        connection.add_option("--replay-proxy", action="store", dest="replay_proxy", type="string", default=self.replay_proxy,
                               help="Proxy to replay with found paths", metavar="PROXY")
         connection.add_option("--scheme", help="Default scheme (for raw request or if there is no scheme in the URL)", action="store",
                               default=self.scheme, dest="scheme", metavar="SCHEME")

--- a/lib/reports/base_report.py
+++ b/lib/reports/base_report.py
@@ -32,7 +32,6 @@ class BaseReport(object):
 
 
 class FileBaseReport(BaseReport):
-
     def __init__(self, host, port, protocol, basePath, output, batch):
         self.output = output
         self.port = port

--- a/lib/reports/base_report.py
+++ b/lib/reports/base_report.py
@@ -21,7 +21,6 @@ import queue
 
 
 class BaseReport(object):
-
     def addPath(selg, path, status, response):
         raise NotImplementedError
 

--- a/lib/reports/csv_report.py
+++ b/lib/reports/csv_report.py
@@ -22,7 +22,6 @@ from lib.reports import *
 
 
 class CSVReport(FileBaseReport):
-
     def addPath(self, path, status, response):
         contentLength = None
 

--- a/lib/reports/csv_report.py
+++ b/lib/reports/csv_report.py
@@ -23,11 +23,8 @@ from lib.reports import *
 
 class CSVReport(FileBaseReport):
     def addPath(self, path, status, response):
-        contentLength = None
-
         try:
             contentLength = int(response.headers["content-length"])
-
         except (KeyError, ValueError):
             contentLength = len(response.body)
 

--- a/lib/reports/json_report.py
+++ b/lib/reports/json_report.py
@@ -23,7 +23,6 @@ from lib.reports import *
 
 
 class JSONReport(FileBaseReport):
-
     def addPath(self, path, status, response):
         contentLength = None
 

--- a/lib/reports/json_report.py
+++ b/lib/reports/json_report.py
@@ -24,11 +24,8 @@ from lib.reports import *
 
 class JSONReport(FileBaseReport):
     def addPath(self, path, status, response):
-        contentLength = None
-
         try:
             contentLength = int(response.headers["content-length"])
-
         except (KeyError, ValueError):
             contentLength = len(response.body)
 

--- a/lib/reports/markdown_report.py
+++ b/lib/reports/markdown_report.py
@@ -21,10 +21,7 @@ import time
 
 
 class MarkdownReport(FileBaseReport):
-
     def addPath(self, path, status, response):
-        contentLength = None
-
         try:
             contentLength = int(response.headers["content-length"])
 

--- a/lib/reports/plain_text_report.py
+++ b/lib/reports/plain_text_report.py
@@ -22,29 +22,24 @@ from lib.utils.file_utils import FileUtils
 import time
 
 
-class PlainTextReport(TailableFileBaseReport):
-
+class PlainTextReport(FileBaseReport):
     def addPath(self, path, status, response):
-        contentLength = None
-        location = None
-
         try:
             contentLength = int(response.headers["content-length"])
-
         except (KeyError, ValueError):
             contentLength = len(response.body)
 
         try:
             location = response.headers["location"]
-        except(KeyError, ValueError):
-            pass
+        except (KeyError, ValueError):
+            location = None
 
         self.storeData((path, status, contentLength, location))
 
     def generate(self):
         result = "Time: {0}\n\n".format(time.ctime())
 
-        for path, status, contentLength, location in self.getPathIterator():
+        for path, status, contentLength, location in self.pathList:
             result += "{0}  ".format(status)
             result += "{0}  ".format(FileUtils.size_human(contentLength).rjust(6, " "))
             result += "{0}://{1}:{2}/".format(self.protocol, self.host, self.port)

--- a/lib/reports/simple_report.py
+++ b/lib/reports/simple_report.py
@@ -19,13 +19,11 @@
 from lib.reports import *
 
 
-class SimpleReport(TailableFileBaseReport):
-
+class SimpleReport(FileBaseReport):
     def generate(self):
         result = ""
 
-        for path, _, _ in self.getPathIterator():
-
+        for path, _, _ in self.pathList:
             result += "{0}://{1}:{2}/".format(self.protocol, self.host, self.port)
             result += (
                 "{0}\n".format(path)

--- a/lib/reports/xml_report.py
+++ b/lib/reports/xml_report.py
@@ -21,7 +21,6 @@ import time
 
 
 class XMLReport(FileBaseReport):
-
     def addPath(self, path, status, response):
         contentLength = None
 

--- a/lib/reports/xml_report.py
+++ b/lib/reports/xml_report.py
@@ -22,8 +22,6 @@ import time
 
 class XMLReport(FileBaseReport):
     def addPath(self, path, status, response):
-        contentLength = None
-
         try:
             contentLength = int(response.headers["content-length"])
 

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,5 +1,5 @@
 certifi>=2020.11.8
 chardet<3.1.0
 urllib3<=1.24.3
-cryptography>1.3.4
+cryptography>=2.8
 PySocks<=1.6.8

